src/testing/jest/README.md

```diff
@@ -0,0 +1,211 @@
+# Jest Support in Stencil
+
+This directory contains support for the Jest testing library.
+
+## Adding Support for a New Version of Jest
+
+The steps for adding support for a new version of Jest can be found below.
+These steps are intended to be followed in order, and be a part of a single pull request.
+**It is HIGHLY recommended that you create a commit for each step in the process.**
+This allows for 'well known' commits that map directly back to these steps to be created, speeding up the review process.
+
+1. In this directory, there are several version-specific directories to support Jest.
+   For example, `jest-27-and-under/` is for Jest v27 and below, `jest-28/` is for Jest v28, etc.
+   To start, create a copy of the entire directory contents for **the latest version** of Jest in this directory, ensuring the version number is in the directory name.
+   
+   If we wanted to add support for Jest v29 and had the following existing directory structure:
+   ```
+   └── src/testing/jest/
+       ├── jest-27-and-under
+       └── jest-28
+   ```
+   Adding support for Jest v29 would begin with making a copy of `jest-28/` (the latest supported version) and naming it `jest-29/`.
+   The end result of this step are two identical copies of the copied directory, with one bearing the name of the new version of Jest:
+   ```
+   └── src/testing/jest/
+       ├── jest-27-and-under
+       ├── jest-28
+       └── jest-29            <- Newly copied version of jest-28, just with a different name
+   ``` 
+
+1. **Commit your changes to help your reviewers out - this makes it easier to review changes that are coming down the line.**
+   Last reminder, please do this after every step.
+1. Update any Jest related dependencies to use the targeted version of Jest.
+   Note: It is up to you to determine/verify the packages needed here.
+   ```bash
+   $ cd src/testing/jest/jest-29
+   $ npm i -D jest@29 @types/jest@29
+   ```
+   Note that all dependencies can simply be dev-dependencies, since they're only used for type checking.
+1. Update the name of the package in the newly created directory to reflect the version of Jest it supports.
+   Do so by replacing the one instance of the package name in `package.json` and two instances in `package-lock-json`.
+   Following the example above, `@stencil/jest-28` would be replaced with `@stencil/jest-29`.
```

Comment:
would it make sense to update the package name before installing the new dependencies? just thinking then the `package-lock.json` would be updated already. also do we maybe want to delete that (the `package-lock.json`) when starting a new adapter?

Reply:
Yeah, that's probably the better thing to do - I tested it out locally and everything worked as expected. Doc updated in https://github.com/ionic-team/stencil/pull/4952/commits/09129f4ebd816f0ec5dee7a416cd5d090231ee5f